lib/data/external/AwsClient.js

```diff
@@ -158,6 +158,39 @@ class AwsClient {
         });
     }
 
+    listParts(key, uploadId, bucket, partNumberMarker, maxParts, log,
+    callback) {
+        const awsBucket = this._awsBucketName;
+        const awsKey = this._createAwsKey(bucket, key, this._bucketMatch);
+        const params = { Bucket: awsBucket, Key: awsKey, UploadId: uploadId,
+            PartNumberMarker: partNumberMarker, MaxParts: maxParts };
+        return this._client.listParts(params, (err, partList) => {
+            if (err) {
+                log.error('err from data backend on listPart',
+                { error: err.message, dataStoreName: this._dataStoreName });
+                return callback(errors.InternalError);
+            }
+            // build storedParts object to mimic Scality S3 backend returns
+            const storedParts = {};
+            storedParts.IsTruncated = partList.IsTruncated;
+            storedParts.Contents = [];
+            storedParts.Contents = partList.Parts.map(item => {
```

Comment:
It would be better if we didn't have to loop through these items twice but I suppose this is less messy code-wise.  We could have the create xml function have an if clause where if it's from aws, the xml creation is different so there is one pass through. What do you think?